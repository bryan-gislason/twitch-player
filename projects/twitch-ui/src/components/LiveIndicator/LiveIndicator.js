import { useTranslation } from "@twitch-player/i18n";
import { useTheme } from "@twitch-player/themes/dist/hooks";
import PropTypes from "prop-types";
import React, { memo } from "react";
import { StyleSheet, View } from "react-native";
import Text from "../Text";

const LiveIndicator = ({ testID = "live-indicator" }) => {
  const { t } = useTranslation();
  const [{ colors, layout, typography }] = useTheme();

  const fontSize = typography.sizeTertiary;
  const dotSize = Math.round(fontSize * 0.75);

  const rootStyle = [
    styles.root,
    {
      backgroundColor: colors.liveIndicatorBackground,
      borderRadius: layout.borderRadiusSmall,
      padding: layout.gapSmall,
      margin: layout.gapMedium,
    },
  ];
  const dotStyle = [
    styles.dot,
    {
      backgroundColor: colors.liveIndicatorDot,
      height: dotSize,
      marginEnd: layout.gapSmall,
      width: dotSize,
    },
  ];
  const textStyle = [
    styles.text,
    {
      color: colors.liveIndicator,
      fontSize,
    },
  ];

  return (
    <View aria-hidden={true} style={rootStyle} testID={testID}>
      <View style={dotStyle} />
      <Text style={textStyle}>{t("components.item.liveIndicator")}</Text>
    </View>
  );
};

LiveIndicator.propTypes = {
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    alignItems: "center",
    flexDirection: "row",
    width: "max-content",
  },
  dot: {
    borderRadius: "50%",
    display: "block",
  },
  text: {
    lineHeight: "1",
    textTransform: "uppercase",
  },
});

export default memo(LiveIndicator);
