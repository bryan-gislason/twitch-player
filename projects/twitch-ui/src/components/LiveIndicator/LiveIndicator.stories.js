import { storiesOf } from "@storybook/react";
import React from "react";
import LiveIndicator from ".";

storiesOf("Components|LiveIndicator", module).add("default", () => (
  <LiveIndicator />
));
