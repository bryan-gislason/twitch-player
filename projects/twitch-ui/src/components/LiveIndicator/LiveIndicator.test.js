import React from "react";
import { render } from "../../testing";
import LiveIndicator from "./LiveIndicator";

describe("LiveIndicator", () => {
  let instance;

  beforeEach(() => {
    instance = render(<LiveIndicator />);
  });

  it("renders a live indicator", () => {
    expect(instance.getByTestId("live-indicator")).toMatchSnapshot();
  });
});
