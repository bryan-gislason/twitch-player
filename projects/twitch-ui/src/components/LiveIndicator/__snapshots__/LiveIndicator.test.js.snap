// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveIndicator renders a live indicator 1`] = `
<div
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-18u37iz rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-position-bnwqim rn-width-1jaylin rn-zIndex-1lgpqti"
  data-testid="live-indicator"
  style="background-color: rgba(0, 0, 0, 0.6); border-top-left-radius: 2px; border-top-right-radius: 2px; border-bottom-right-radius: 2px; border-bottom-left-radius: 2px; margin: 10px 10px 10px 10px; padding: 5px 5px 5px 5px;"
>
  <div
    aria-hidden="true"
    class="rn-alignItems-1oszu61 rn-borderTopLeftRadius-wlx8yz rn-borderTopRightRadius-iy595e rn-borderBottomRightRadius-1fm23du rn-borderBottomLeftRadius-1p9x04s rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1adg3ll rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    style="background-color: rgb(236, 19, 19); height: 10px; margin-right: 5px; width: 10px;"
  />
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-eljoum rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-textTransform-tsynxw rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
    style="color: rgb(250, 249, 250); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 13px;"
  >
    Live
  </div>
</div>
`;
